frontend/src/components/Profile/MyMatchHistory.vue

```diff
@@ -17,29 +35,104 @@ import axios from 'axios';
 import { defineComponent } from 'vue';
 import { mapGetters } from 'vuex';
 
+interface Match {
+  idOther: number,
+  tabMatchs: { status: string, homeId: number, awayId: number, matchId: number, usernameOne: string, usernameTwo: string, matchNum: number, } [],
+  idMatch: number,
+  dialog: boolean,
+  username: string,
+}
+
 export default defineComponent({
-  data: () => ({
-    idOther: 0,
-  }),
+  data(): Match {
+    return {
+      idOther: 0,
+      tabMatchs: [],
+      idMatch: 0,
+      dialog: false,
+      username: '',
+    };
+  },
   methods: {
     ...mapGetters(['id']),
+    async setWinner(id: number) {
+      let response = await axios.get('/users/' + id + '/matches');
+      for (let i: number = 0 ; i < response.data.length ; i++) {
+        this.idMatch = response.data[i].id;
+        let response2 = await axios.get('/matches/' + this.idMatch)
+        this.tabMatchs.push({
+          status: response2.data.status,
+          homeId: response2.data.homeId,
+          awayId: response2.data.awayId,
+          matchId: response2.data.id,
+          usernameOne: '',
+          usernameTwo: '',
+          matchNum: 0,
+        });
+      }
+      for (let i: number = 0 ; i < this.tabMatchs.length ; i++) {
```

Comment:
This could also be way better is we changed the dto so that it would process and indicates who's the winner and everything, but again, given the time we have left this is good enough for me.

Reply:
I agree. But I also agree on the fact that we don't have much time. I take note of it, to be able to return to it if we have time.